language-bindings/rust/build.rs

```diff
@@ -0,0 +1,30 @@
+extern crate bindgen;
+extern crate cmake;
+
+use cmake::Config;
+use std::{env, path::PathBuf};
+
+fn main() {
+    let mut config = Config::new("../..");
+    let dst = config
+        .generator("Unix Makefiles")
+        .define("WAMR_BUILD_TARGET", env::var("TARGET").unwrap().split("-").next().unwrap().to_uppercase())
+        .build();
```

Comment:
please feel free to correct me.

becuase we can't ask crate users to re-compile,
I guess we need to prepare a compilation options combination here in order to define 
- which features we are going to provide in the final publish crate 
- which running mode we are going to provide in the final publish crate.

Reply:
The build.rs is run during the crate installation. It is not uncommon to compile dependencies here.

For the execution mode (like interpreted, AOT, JIT) we can hopefully use feature flags on the crate.

For environments where it is not possible to compile wamr, we could try to "vendor" wamr and include it in the crate.